<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential expression with DESeq2 - Dataset 2 | Differential Expression Tutorial</title>
  <meta name="description" content="Chapter 11 Differential expression with DESeq2 - Dataset 2 | Differential Expression Tutorial" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential expression with DESeq2 - Dataset 2 | Differential Expression Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential expression with DESeq2 - Dataset 2 | Differential Expression Tutorial" />
  
  
  

<meta name="author" content="BCB420 / Ruth Isserlin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="edger_dataset2.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BCB420 Lecture </a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#packages"><i class="fa fa-check"></i><b>1.1</b> Packages</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#reproducibility"><i class="fa fa-check"></i><b>1.2</b> Reproducibility</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="get-data.html"><a href="get-data.html"><i class="fa fa-check"></i><b>2</b> Get the data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="get-data.html"><a href="get-data.html#download-the-geo-supplementary-files"><i class="fa fa-check"></i><b>2.1</b> Download the GEO supplementary file(s)</a></li>
<li class="chapter" data-level="2.2" data-path="get-data.html"><a href="get-data.html#read-the-featurecounts-matrix"><i class="fa fa-check"></i><b>2.2</b> Read the FeatureCounts matrix</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="get-data.html"><a href="get-data.html#build-a-counts-matrix-genes-samples"><i class="fa fa-check"></i><b>2.2.1</b> Build a counts matrix (genes × samples)</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="get-data.html"><a href="get-data.html#create-metadata-from-sample-names"><i class="fa fa-check"></i><b>2.3</b> Create metadata from sample names</a></li>
<li class="chapter" data-level="2.4" data-path="get-data.html"><a href="get-data.html#identifier-mapping"><i class="fa fa-check"></i><b>2.4</b> Identifier mapping</a></li>
<li class="chapter" data-level="2.5" data-path="get-data.html"><a href="get-data.html#save-intermediate-objects"><i class="fa fa-check"></i><b>2.5</b> Save intermediate objects</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="qc.html"><a href="qc.html"><i class="fa fa-check"></i><b>3</b> QC, filtering and identifier mapping</a>
<ul>
<li class="chapter" data-level="3.1" data-path="qc.html"><a href="qc.html#filter-lowly-expressed-genes"><i class="fa fa-check"></i><b>3.1</b> Filter lowly expressed genes</a></li>
<li class="chapter" data-level="3.2" data-path="qc.html"><a href="qc.html#tmm-normalization"><i class="fa fa-check"></i><b>3.2</b> TMM normalization</a></li>
<li class="chapter" data-level="3.3" data-path="qc.html"><a href="qc.html#exploratory-mds-plot"><i class="fa fa-check"></i><b>3.3</b> Exploratory MDS plot</a></li>
<li class="chapter" data-level="3.4" data-path="qc.html"><a href="qc.html#exploratory-mds-plot---coloured-by-cellline"><i class="fa fa-check"></i><b>3.4</b> Exploratory MDS plot - coloured by cellline</a></li>
<li class="chapter" data-level="3.5" data-path="qc.html"><a href="qc.html#log-cpm-for-visualization"><i class="fa fa-check"></i><b>3.5</b> log-CPM for visualization</a></li>
<li class="chapter" data-level="3.6" data-path="qc.html"><a href="qc.html#save-for-downstream-chapters"><i class="fa fa-check"></i><b>3.6</b> Save for downstream chapters</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="limma-voom.html"><a href="limma-voom.html"><i class="fa fa-check"></i><b>4</b> Differential expression with limma-voom</a>
<ul>
<li class="chapter" data-level="4.1" data-path="limma-voom.html"><a href="limma-voom.html#results-tables"><i class="fa fa-check"></i><b>4.1</b> Results tables</a></li>
<li class="chapter" data-level="4.2" data-path="limma-voom.html"><a href="limma-voom.html#volcano-plot-20ctg-vs-nt"><i class="fa fa-check"></i><b>4.2</b> Volcano plot (20CTG vs NT)</a></li>
<li class="chapter" data-level="4.3" data-path="limma-voom.html"><a href="limma-voom.html#save-limma-results"><i class="fa fa-check"></i><b>4.3</b> Save limma results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="edger.html"><a href="edger.html"><i class="fa fa-check"></i><b>5</b> Differential expression with edgeR (QLF)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="edger.html"><a href="edger.html#plot-biological-coefficient-of-variance-plot"><i class="fa fa-check"></i><b>5.1</b> Plot Biological coefficient of variance plot</a></li>
<li class="chapter" data-level="5.2" data-path="edger.html"><a href="edger.html#test-3ctg-vs-nt-and-20ctg-vs-nt"><i class="fa fa-check"></i><b>5.2</b> Test: 3CTG vs NT and 20CTG vs NT</a></li>
<li class="chapter" data-level="5.3" data-path="edger.html"><a href="edger.html#test-20ctg-vs-3ctg"><i class="fa fa-check"></i><b>5.3</b> Test: 20CTG vs 3CTG</a></li>
<li class="chapter" data-level="5.4" data-path="edger.html"><a href="edger.html#extract-full-results"><i class="fa fa-check"></i><b>5.4</b> Extract full results</a></li>
<li class="chapter" data-level="5.5" data-path="edger.html"><a href="edger.html#volcano-plot-of-edger-results"><i class="fa fa-check"></i><b>5.5</b> Volcano plot of EdgeR results</a></li>
<li class="chapter" data-level="5.6" data-path="edger.html"><a href="edger.html#visualize-the-top-hits-as-a-heatmap"><i class="fa fa-check"></i><b>5.6</b> Visualize the top hits as a heatmap</a></li>
<li class="chapter" data-level="5.7" data-path="edger.html"><a href="edger.html#select-top-hits-from-tab_20_vs_nt_annot"><i class="fa fa-check"></i><b>5.7</b> Select top hits from <code>tab_20_vs_nt_annot</code></a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="edger.html"><a href="edger.html#top-n-genes-by-fdr"><i class="fa fa-check"></i><b>5.7.1</b> top N genes by FDR</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="edger.html"><a href="edger.html#make-the-expression-matrix-for-the-heatmap-logcpm-row-z-score"><i class="fa fa-check"></i><b>5.8</b> Make the expression matrix for the heatmap (logCPM + row z-score)</a></li>
<li class="chapter" data-level="5.9" data-path="edger.html"><a href="edger.html#sample-annotation-treatment-cellline-using-set3"><i class="fa fa-check"></i><b>5.9</b> Sample annotation (treatment + cellline) using Set3</a></li>
<li class="chapter" data-level="5.10" data-path="edger.html"><a href="edger.html#draw-the-complexheatmap-heatmap"><i class="fa fa-check"></i><b>5.10</b> 5) Draw the ComplexHeatmap heatmap</a></li>
<li class="chapter" data-level="5.11" data-path="edger.html"><a href="edger.html#save-edger-results"><i class="fa fa-check"></i><b>5.11</b> Save edgeR results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="deseq2.html"><a href="deseq2.html"><i class="fa fa-check"></i><b>6</b> Differential expression with DESeq2</a>
<ul>
<li class="chapter" data-level="6.1" data-path="deseq2.html"><a href="deseq2.html#results"><i class="fa fa-check"></i><b>6.1</b> Results</a></li>
<li class="chapter" data-level="6.2" data-path="deseq2.html"><a href="deseq2.html#volcano-redup-bluedown"><i class="fa fa-check"></i><b>6.2</b> Volcano (red=up, blue=down)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html"><i class="fa fa-check"></i><b>7</b> Get the data - Dataset 2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html#download-the-geo-supplementary-files-1"><i class="fa fa-check"></i><b>7.1</b> Download the GEO supplementary file(s)</a></li>
<li class="chapter" data-level="7.2" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html#read-the-featurecounts-matrix-1"><i class="fa fa-check"></i><b>7.2</b> Read the FeatureCounts matrix</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html#build-a-counts-matrix-genes-samples-1"><i class="fa fa-check"></i><b>7.2.1</b> Build a counts matrix (genes × samples)</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html#create-metadata-from-sample-names-1"><i class="fa fa-check"></i><b>7.3</b> Create metadata from sample names</a></li>
<li class="chapter" data-level="7.4" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html#identifier-mapping-1"><i class="fa fa-check"></i><b>7.4</b> Identifier mapping</a></li>
<li class="chapter" data-level="7.5" data-path="get-data_dataset2.html"><a href="get-data_dataset2.html#save-intermediate-objects-1"><i class="fa fa-check"></i><b>7.5</b> Save intermediate objects</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="qc_dataset2.html"><a href="qc_dataset2.html"><i class="fa fa-check"></i><b>8</b> QC, filtering and identifier mapping - Dataset 2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="qc_dataset2.html"><a href="qc_dataset2.html#load-in-the-data"><i class="fa fa-check"></i><b>8.1</b> Load in the data</a></li>
<li class="chapter" data-level="8.2" data-path="qc_dataset2.html"><a href="qc_dataset2.html#simplify-this-model"><i class="fa fa-check"></i><b>8.2</b> Simplify this model</a></li>
<li class="chapter" data-level="8.3" data-path="qc_dataset2.html"><a href="qc_dataset2.html#filter-lowly-expressed-genes-1"><i class="fa fa-check"></i><b>8.3</b> Filter lowly expressed genes</a></li>
<li class="chapter" data-level="8.4" data-path="qc_dataset2.html"><a href="qc_dataset2.html#tmm-normalization-1"><i class="fa fa-check"></i><b>8.4</b> TMM normalization</a></li>
<li class="chapter" data-level="8.5" data-path="qc_dataset2.html"><a href="qc_dataset2.html#exploratory-mds-plot-1"><i class="fa fa-check"></i><b>8.5</b> Exploratory MDS plot</a></li>
<li class="chapter" data-level="8.6" data-path="qc_dataset2.html"><a href="qc_dataset2.html#exploratory-mds-plot---coloured-by-patient"><i class="fa fa-check"></i><b>8.6</b> Exploratory MDS plot - coloured by patient</a></li>
<li class="chapter" data-level="8.7" data-path="qc_dataset2.html"><a href="qc_dataset2.html#log-cpm-for-visualization-1"><i class="fa fa-check"></i><b>8.7</b> log-CPM for visualization</a></li>
<li class="chapter" data-level="8.8" data-path="qc_dataset2.html"><a href="qc_dataset2.html#save-for-downstream-chapters-1"><i class="fa fa-check"></i><b>8.8</b> Save for downstream chapters</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="limma-voom_dataset2.html"><a href="limma-voom_dataset2.html"><i class="fa fa-check"></i><b>9</b> Differential expression with limma-voom - Dataset 2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="limma-voom_dataset2.html"><a href="limma-voom_dataset2.html#results-tables-1"><i class="fa fa-check"></i><b>9.1</b> Results tables</a></li>
<li class="chapter" data-level="9.2" data-path="limma-voom_dataset2.html"><a href="limma-voom_dataset2.html#volcano-plot-0-vs-rest"><i class="fa fa-check"></i><b>9.2</b> Volcano plot (0 vs rest)</a></li>
<li class="chapter" data-level="9.3" data-path="limma-voom_dataset2.html"><a href="limma-voom_dataset2.html#save-limma-results-1"><i class="fa fa-check"></i><b>9.3</b> Save limma results</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="edger_dataset2.html"><a href="edger_dataset2.html"><i class="fa fa-check"></i><b>10</b> Differential expression with edgeR (QLF) - Dataset 2</a>
<ul>
<li class="chapter" data-level="10.1" data-path="edger_dataset2.html"><a href="edger_dataset2.html#plot-biological-coefficient-of-variance-plot-1"><i class="fa fa-check"></i><b>10.1</b> Plot Biological coefficient of variance plot</a></li>
<li class="chapter" data-level="10.2" data-path="edger_dataset2.html"><a href="edger_dataset2.html#test-status-0-vs-status-1"><i class="fa fa-check"></i><b>10.2</b> Test: status 0 vs status 1</a></li>
<li class="chapter" data-level="10.3" data-path="edger_dataset2.html"><a href="edger_dataset2.html#extract-full-results-1"><i class="fa fa-check"></i><b>10.3</b> Extract full results</a></li>
<li class="chapter" data-level="10.4" data-path="edger_dataset2.html"><a href="edger_dataset2.html#volcano-plot-of-edger-results-1"><i class="fa fa-check"></i><b>10.4</b> Volcano plot of EdgeR results</a></li>
<li class="chapter" data-level="10.5" data-path="edger_dataset2.html"><a href="edger_dataset2.html#try-and-run-glmfit"><i class="fa fa-check"></i><b>10.5</b> Try and run glmFit</a></li>
<li class="chapter" data-level="10.6" data-path="edger_dataset2.html"><a href="edger_dataset2.html#test-status-0-vs-status-1-1"><i class="fa fa-check"></i><b>10.6</b> Test: status 0 vs status 1</a></li>
<li class="chapter" data-level="10.7" data-path="edger_dataset2.html"><a href="edger_dataset2.html#save-edger-results-1"><i class="fa fa-check"></i><b>10.7</b> Save edgeR results</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="deseq2_dataset2.html"><a href="deseq2_dataset2.html"><i class="fa fa-check"></i><b>11</b> Differential expression with DESeq2 - Dataset 2</a>
<ul>
<li class="chapter" data-level="11.1" data-path="deseq2_dataset2.html"><a href="deseq2_dataset2.html#results-1"><i class="fa fa-check"></i><b>11.1</b> Results</a></li>
<li class="chapter" data-level="11.2" data-path="deseq2_dataset2.html"><a href="deseq2_dataset2.html#volcano-redup-bluedown-1"><i class="fa fa-check"></i><b>11.2</b> Volcano (red=up, blue=down)</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differential Expression Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="deseq2_dataset2" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential expression with DESeq2 - Dataset 2<a href="deseq2_dataset2.html#deseq2_dataset2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>DESeq2 performs differential expression analysis for RNA-seq <strong>count</strong> data.</p>
<p>We are assuming here that the counts data are estimated counts exported from salmon - but that is not guarenteed.</p>
<p>DESeq requires count data - specifically integer count data (not an estimated version). We will round these numbers but this is not ideal. You should ideally have count data.</p>
<p>If you have TPM, (or RPKM, or FPKM) your real only route is to use limma as both edgeR and DESEq2 require raw counts.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="deseq2_dataset2.html#cb105-1" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/d2_qc_objects.rds&quot;</span>)</span>
<span id="cb105-2"><a href="deseq2_dataset2.html#cb105-2" tabindex="-1"></a>dge_f <span class="ot">&lt;-</span> obj<span class="sc">$</span>dge_f</span>
<span id="cb105-3"><a href="deseq2_dataset2.html#cb105-3" tabindex="-1"></a>meta <span class="ot">&lt;-</span> obj<span class="sc">$</span>meta</span>
<span id="cb105-4"><a href="deseq2_dataset2.html#cb105-4" tabindex="-1"></a></span>
<span id="cb105-5"><a href="deseq2_dataset2.html#cb105-5" tabindex="-1"></a>design <span class="ot">=</span> obj<span class="sc">$</span>design</span>
<span id="cb105-6"><a href="deseq2_dataset2.html#cb105-6" tabindex="-1"></a></span>
<span id="cb105-7"><a href="deseq2_dataset2.html#cb105-7" tabindex="-1"></a><span class="do">## but DESeq requires the formula and not the model.matrix object</span></span>
<span id="cb105-8"><a href="deseq2_dataset2.html#cb105-8" tabindex="-1"></a><span class="do">## below we put the formula in directly </span></span>
<span id="cb105-9"><a href="deseq2_dataset2.html#cb105-9" tabindex="-1"></a></span>
<span id="cb105-10"><a href="deseq2_dataset2.html#cb105-10" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb105-11"><a href="deseq2_dataset2.html#cb105-11" tabindex="-1"></a></span>
<span id="cb105-12"><a href="deseq2_dataset2.html#cb105-12" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">round</span>(dge_f<span class="sc">$</span>counts)</span>
<span id="cb105-13"><a href="deseq2_dataset2.html#cb105-13" tabindex="-1"></a></span>
<span id="cb105-14"><a href="deseq2_dataset2.html#cb105-14" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb105-15"><a href="deseq2_dataset2.html#cb105-15" tabindex="-1"></a>  <span class="at">countData =</span> count_mat,</span>
<span id="cb105-16"><a href="deseq2_dataset2.html#cb105-16" tabindex="-1"></a>  <span class="at">colData =</span> <span class="fu">as.data.frame</span>(meta),</span>
<span id="cb105-17"><a href="deseq2_dataset2.html#cb105-17" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> status </span>
<span id="cb105-18"><a href="deseq2_dataset2.html#cb105-18" tabindex="-1"></a>)</span>
<span id="cb105-19"><a href="deseq2_dataset2.html#cb105-19" tabindex="-1"></a></span>
<span id="cb105-20"><a href="deseq2_dataset2.html#cb105-20" tabindex="-1"></a></span>
<span id="cb105-21"><a href="deseq2_dataset2.html#cb105-21" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<div id="results-1" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Results<a href="deseq2_dataset2.html#results-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="deseq2_dataset2.html#cb106-1" tabindex="-1"></a>res_0_vs_1 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds,<span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;status&quot;</span>,<span class="st">&quot;0&quot;</span>,<span class="st">&quot;1&quot;</span>))</span>
<span id="cb106-2"><a href="deseq2_dataset2.html#cb106-2" tabindex="-1"></a></span>
<span id="cb106-3"><a href="deseq2_dataset2.html#cb106-3" tabindex="-1"></a><span class="fu">summary</span>(res_20_vs_nt)</span></code></pre></div>
<pre><code>## 
## out of 14704 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 541, 3.7%
## LFC &lt; 0 (down)     : 225, 1.5%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 4)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
</div>
<div id="volcano-redup-bluedown-1" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Volcano (red=up, blue=down)<a href="deseq2_dataset2.html#volcano-redup-bluedown-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="deseq2_dataset2.html#cb108-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb108-2"><a href="deseq2_dataset2.html#cb108-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb108-3"><a href="deseq2_dataset2.html#cb108-3" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb108-4"><a href="deseq2_dataset2.html#cb108-4" tabindex="-1"></a></span>
<span id="cb108-5"><a href="deseq2_dataset2.html#cb108-5" tabindex="-1"></a>vol <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res_0_vs_1) <span class="sc">%&gt;%</span></span>
<span id="cb108-6"><a href="deseq2_dataset2.html#cb108-6" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb108-7"><a href="deseq2_dataset2.html#cb108-7" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb108-8"><a href="deseq2_dataset2.html#cb108-8" tabindex="-1"></a>    <span class="at">neglog10p =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue),</span>
<span id="cb108-9"><a href="deseq2_dataset2.html#cb108-9" tabindex="-1"></a>    <span class="at">direction =</span> <span class="fu">case_when</span>(</span>
<span id="cb108-10"><a href="deseq2_dataset2.html#cb108-10" tabindex="-1"></a>      <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Up&quot;</span>,</span>
<span id="cb108-11"><a href="deseq2_dataset2.html#cb108-11" tabindex="-1"></a>      <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Down&quot;</span>,</span>
<span id="cb108-12"><a href="deseq2_dataset2.html#cb108-12" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Not sig&quot;</span></span>
<span id="cb108-13"><a href="deseq2_dataset2.html#cb108-13" tabindex="-1"></a>    )</span>
<span id="cb108-14"><a href="deseq2_dataset2.html#cb108-14" tabindex="-1"></a>  )</span>
<span id="cb108-15"><a href="deseq2_dataset2.html#cb108-15" tabindex="-1"></a></span>
<span id="cb108-16"><a href="deseq2_dataset2.html#cb108-16" tabindex="-1"></a><span class="fu">ggplot</span>(vol, <span class="fu">aes</span>(log2FoldChange, neglog10p)) <span class="sc">+</span></span>
<span id="cb108-17"><a href="deseq2_dataset2.html#cb108-17" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> direction), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb108-18"><a href="deseq2_dataset2.html#cb108-18" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="at">Up =</span> <span class="st">&quot;red&quot;</span>, <span class="at">Down =</span> <span class="st">&quot;blue&quot;</span>, <span class="st">`</span><span class="at">Not sig</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;grey70&quot;</span>)) <span class="sc">+</span></span>
<span id="cb108-19"><a href="deseq2_dataset2.html#cb108-19" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb108-20"><a href="deseq2_dataset2.html#cb108-20" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;DESeq2: 0 vs 1&quot;</span>, <span class="at">x =</span> <span class="st">&quot;log2 FC&quot;</span>, <span class="at">y =</span> <span class="st">&quot;-log10(p)&quot;</span>, <span class="at">color =</span> <span class="cn">NULL</span>)</span></code></pre></div>
<p><img src="DE_tutorial_files/figure-html/unnamed-chunk-65-1.png" width="672" /></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="deseq2_dataset2.html#cb109-1" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="fu">list</span>(<span class="at">meta =</span> meta, </span>
<span id="cb109-2"><a href="deseq2_dataset2.html#cb109-2" tabindex="-1"></a>             <span class="at">dds =</span> dds, </span>
<span id="cb109-3"><a href="deseq2_dataset2.html#cb109-3" tabindex="-1"></a>             <span class="at">res_20_vs_nt =</span> res_20_vs_nt,</span>
<span id="cb109-4"><a href="deseq2_dataset2.html#cb109-4" tabindex="-1"></a>             <span class="at">res_3_vs_nt =</span> res_3_vs_nt, </span>
<span id="cb109-5"><a href="deseq2_dataset2.html#cb109-5" tabindex="-1"></a>             <span class="at">res_20_vs_3 =</span> res_20_vs_3),</span>
<span id="cb109-6"><a href="deseq2_dataset2.html#cb109-6" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">&quot;data/d2_deseq2_results.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="edger_dataset2.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["DE_tutorial.pdf", "DE_tutorial.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
